src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -296,11 +296,78 @@ private OpenApiRequestBody GetFormRequestBody(IList<ApiRequestFormat> supportedR
             Content = new Dictionary<string, OpenApiMediaType>()
         };
 
-        // Forms are represented as objects with properties for each form field.
         var schema = new OpenApiSchema { Type = "object", Properties = new Dictionary<string, OpenApiSchema>() };
-        foreach (var parameter in formParameters)
+        // Group form parameters by their parameter name because MVC explodes form parameters that are bound from the
+        // same model instance into separate parameters, while minimal APIs does not.
+        //
+        // public record Todo(int Id, string Title, bool Completed, DateTime CreatedAt)
+        // public void PostMvc([FromForm] Todo person) { }
+        // app.MapGet("/form-todo", ([FromForm] Todo todo) => Results.Ok(todo));
+        //
+        // In the example above, MVC will bind four separate arguments to the Todo model while minimal APIs will
+        // bind a single Todo model instance to the todo parameter. Grouping by name allows us to handle both cases.
+        var groupedFormParameters = formParameters.GroupBy(parameter => parameter.ParameterDescriptor.Name).ToList();
+        // If there is only one real parameter derive from the form body, then set it directly in the schema.
+        if (groupedFormParameters is [var groupedParameter])
         {
-            schema.Properties[parameter.Name] = _componentService.GetOrCreateSchema(parameter.Type);
+            // If the group contains one element, then we're dealing with a minimal API scenario where a
+            // single form parameter produces a single API description parameter.
+            if (groupedParameter.Count() == 1)
```

Comment:
I assume it's _incidentally_ true that it's a minimal API and that this behavior makes sense any time there's a single parameter in the group?

Reply:
Related: What happens if it's an MVC parameter and the object has only one property?  Would that show up the same way?

Reply:
Good observation! I've hardened the case for the model with one property scenario.